### PR TITLE
TEP-0147: introduce feature flag to guard artifacts feature

### PR DESCRIPTION
**Pull request by ericzzzzzzz**
# Changes

fixes:  #7700 
 - We introduce a feature flag to guard artifacts feature details see https://github.com/tektoncd/community/blob/main/teps/0147-tekton-artifacts-phase1.md 
 - This pr takes the same approach as https://github.com/tektoncd/pipeline/pull/7277

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Introduces a feature flag enable-artifacts.
```

### CODE REVIEW

**Comment by ericzzzzzzz** [comment]
@chitrangpatel

**Comment by chitrangpatel** [comment]
@ericzzzzzzz I would put in a release note to something like:

```
WIP: Introduces a feature flag enable-artifacts.
```

**Comment by chitrangpatel** [comment]
cc @tektoncd/core-maintainers PTAL. This PR starts the work on Tekton Artifacts phase 1.

File: test/featureflags.go [hunk 1]
```diff
@@ -117,6 +117,7 @@ func getFeatureFlagsBaseOnAPIFlag(t *testing.T) *config.FeatureFlags {
 		"enable-step-actions":          "true",
 		"enable-cel-in-whenexpression": "true",
 		"enable-param-enum":            "true",
+		"enable-artifacts":             "true",
```
Comment (JeromeJu) [inline test/featureflags.go:120]: Thanks for updating this, would you mind also adding it to https://github.com/tektoncd/pipeline/blob/a9d8a9b1c7459b64190bd0f73014cee6e56e2df1/test/per_feature_flags_test.go#L50?

Comment (ericzzzzzzz) [reply test/featureflags.go:120]: thanks, added

**Review by JeromeJu — APPROVED** [review]

**Pull request merged**